openedx/core/djangoapps/user_api/learner_skill_levels/views.py

```diff
@@ -93,6 +93,11 @@ def get(self, request, job_id):
         """
         # get top categories for the given job
         job_skill_categories = get_top_skill_categories_for_job(job_id)
+        if not job_skill_categories:
+            return Response(
```

Comment:
you can just use `raise Http404`. Django automatically returns the appropriate error.